### PR TITLE
Issue-4977 Fix bug with new Apple provisioning profile

### LINKED ISSUE #4977: Bug with new Apple provisioning profile
**Describe the bug (REQUIRED)**
New Apple provisioning profile contains an array in `keychain-access-groups` , bob.jar split it up to two separate `keychain-access-groups` which Is the reason of the crash when start app on a device.

More info is available here:
https://forum.defold.com/t/trouble-building-for-ios/65685

**To Reproduce (REQUIRED)**
1. Create new provisioning profile on Apple portal
2. Make sure you have an array in `entitlements` for `keychain-access-groups`, just run [this](https://gist.github.com/jessedc/9383378) command for provisioning
3. Build project and try to run on device

**Expected behavior (REQUIRED)**
App should work fine

**Defold version (REQUIRED):**
  - Version 1.2.170

**Platforms (REQUIRED):**
 - Platforms:  iOS, maybe macOS (I didn't check)
 - OS: tested on iOS 13.5.1
 - Device: tested on iPhone X

**Workaround (OPTIONAL):**
Here: https://forum.defold.com/t/trouble-building-for-ios/65685/8?u=agulev

### PR DESCRIPTION
**Pull request by AGulev**
fix https://github.com/defold/defold/issues/4977

The version of `commons-configuration` we use works wrong with arrays and saves them as separate blocks.
We need to update `commons-configuration` to 2nd version where the bug was fixed. Here is original issue: https://issues.apache.org/jira/browse/CONFIGURATION-427?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=12928591#action_12928591

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java [hunk 1]
```diff
@@ -29,8 +32,11 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-import org.apache.commons.configuration.ConfigurationException;
-import org.apache.commons.configuration.plist.XMLPropertyListConfiguration;
+import org.apache.commons.configuration2.io.FileLocator;
+import org.apache.commons.configuration2.io.FileLocator.FileLocatorBuilder;
+import org.apache.commons.configuration2.io.FileLocatorUtils;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.configuration2.plist.XMLPropertyListConfiguration;
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java:39]: I think it'd be good to have some explanation of what the bug is, and how it's related to updating the java packages?

Comment (AGulev) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java:39]: Hm, I thought it's clear from the issue I created. Ok, I'll add a bit more info in PR too

Comment (AGulev) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java:39]: I added a short explanation in the PR description. Thank you.

**Review by matgis — APPROVED** [review]
Let's get this merged!

**Comment by britzl** [comment]
We're currently stuck waiting for Apple to finish our developer enrollment. A real miscalculation on my part...

**Comment by AGulev** [comment]
> We're currently stuck waiting for Apple to finish our developer enrollment. A real miscalculation on my part...

This branch is ready for merge anyway.

**Pull request merged**